src/Riok.Mapperly/Descriptors/MappingBodyBuilders/BuilderContext/MembersMappingBuilderContext.cs

```diff
@@ -44,9 +46,19 @@ protected MembersMappingBuilderContext(MappingBuilderContext builderContext, T m
 
         MemberConfigsByRootTargetName = GetMemberConfigurations();
 
+        var mappedSourceMemberNames = MemberConfigsByRootTargetName.Values
+            .SelectMany(v => v.Select(s => s.Source.Path.First()))
+            .Distinct()
+            .ToList();
```

Comment:
Couldn't you use `ToHashSet` instead of `Distinct` + `ToList`? The next lines could probably even be combined with this:
```c#
_mappedAndIgnoredSourceMemberNames = MemberConfigsByRootTargetName.Values
            .SelectMany(v => v.Select(s => s.Source.Path.First()))
            .ToHashSet();
_mappedAndIgnoredSourceMemberNames.IntersectWith(IgnoredSourceMemberNames);

_mappedAndIgnoredTargetMemberNames = new HashSet<string>(ignoredTargetMemberNames);
_mappedAndIgnoredTargetMemberNames.IntersectWith(MemberConfigsByRootTargetName.Keys);
```

Couldn't they? This should lead to a better performance than the enumerable based intersect. For the target member names we could even create a custom hash set extension method which would return an enumerable of the actual removed items (which is the intersection and would therefore exactly be the contents of `_mappedAndIgnoredTargetMemberNames`).

Reply:
Thanks for the tip: done in [c345994](https://github.com/riok/mapperly/pull/708/commits/c3459946568be35b163b7e2fe5352718561d0b4d)
PS: I stopped writing the extension method when the clearest names I could come up with looked too messy: `IntersectWithAndGetExcluded` / `IntersectWithAndGetDifference` / `IntersectionAndDifference`